### PR TITLE
Added navigation between multiple sign prompts and reject all sign prompts

### LINKED ISSUE #16848: [Bug]: Cannot navigate through multiple Sign prompts for Signed Typed Data `V3` and `V4`
### Describe the bug

**Problem**: we cannot navigate through multiple Sign prompts for Signed Typed Data V3 and V4. The expected behaviour is the one we see for Signed Type Data, Personal Sign and Eth Sign.

https://user-images.githubusercontent.com/54408225/206234177-845cbb7e-8454-47c6-bcbd-4daedb292ebd.mp4

### Steps to reproduce

1. Connect to Test Dapp https://metamask.github.io/test-dapp/
2. Click 3 times on Sign for Typed Data V4 -- see you cannot navigate through the multiple prompts nor Reject them all at once
3. Reject one by one
4. Click 3 times on Sign for Typed Data V3 -- see you cannot navigate through the multiple prompts nor Reject them all at once

### Error messages or log output

_No response_

### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by VSaric**
## Explanation

With this fix, the users should be able to navigate between multiple sign prompts back and forth and should be able to reject all sign prompts at once for `Eth Sign`, `Personal Sign`, `SignTypedData`, `SignTypedData_v3` and `SignTypedData_v4`. Also we are using a single component for navigation which will show both messages and transactions requests.

* Fixes #16848 
* Blocked by #16822 (ConfirmPageContainerNavigation refactor as it mentioned [here](https://github.com/MetaMask/metamask-extension/pull/16822#discussion_r1047452658))

### Before

https://user-images.githubusercontent.com/92527393/211017881-9149191c-9a8f-47bf-bb7a-33fa7b1c6bb8.mp4

#### popup view

https://user-images.githubusercontent.com/92527393/211017166-8b848b77-a2c4-4c61-a154-ecd08cc7e93e.mov

#### expanded view

https://user-images.githubusercontent.com/92527393/211015295-6cc2c555-5265-49a3-857b-a8f0b64cd19a.mov

### Using a single component for navigation which will show both messages and transactions requests and visa versa

https://user-images.githubusercontent.com/92527393/214558962-d9645d5c-9e76-4055-997d-074107f4c67e.mov

## Manual Testing Steps

1. Connect to Test Dapp https://metamask.github.io/test-dapp/
2. Click multiple times on `Eth Sign` or `Personal Sign` or `SignTypedData` or `SignTypedData_v3`or `SignTypedData_v4`.
3. The users should be able to navigate between multiple sign prompts back and forth and should be able to reject all sign prompts at once

### Using a single component for navigation which will show both messages and transactions requests and visa versa

1. Connect to Test Dapp https://metamask.github.io/test-dapp/
2. Click multiple times on `Piggy bank contract` section (`Deploy Contract`) or `Send Tokens` section  (`Create Token`) or `Send Eth` section (`Send Legacy Transaction` or `Send EIP 1559 Transaction`) and than click multiple times on `Eth Sign` or `Personal Sign` or `SignTypedData` or `SignTypedData_v3`or `SignTypedData_v4`.
3. The users should be able to navigate between multiple sign prompts and transactions back and forth

### CODE REVIEW

**Comment by dzfjz** [comment]
Verified by QA

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by jpuri — CHANGES_REQUESTED** [review]
Changes in [confirm-page-container-navigation.component.js](https://github.com/MetaMask/metamask-extension/pull/17093/files#diff-b003f2a56b8e58f8ee673318a99652e8a205c3f1bd57b8b9384606d77b595dc8) can cause unapproved signature request to be opened from transaction page and visa versa.

An option could be creating separate component for signatures. This new component and `confirm-page-container-navigation.component.js` can be refactored to use same dumb component for UI rendering.

**Comment by seaona** [comment]
Some behaviour to double-check related to mixing Signatures with Sends/Deploys tx's: after prompting for a couple of signatures without accepting them, if the Dapp now sends Send/Deploy tx's requests, MM starts to behave strangely. Issues:

- after some pending Signatures, you try a couple of Sends, and the top navigation displays now the number of pending Sends, but with the Signatures screens. Imagine you had 3 pending Signatures and 2 pending Sends: you'll see navigation for the Sends (2), but the screen for Signatures
- if you then navigate to the Next page, you are now seeing the pending Sends with their navigation and the Signatures won't appears anymore until you reject/accept the pending Sends.

https://user-images.githubusercontent.com/54408225/213418689-77acc705-a7b6-4336-9883-642702b65355.mp4


I am not sure what should be the expect behaviour though:
- are we aiming to consider all Dapp API calls into one navigation flow? I think that would be good for the usability, but not sure how complex might be to implement, depending on how the code is at the moment
- if we want to separate the different Dapp requests, we should fix the navigation mix and the clicking Next, making switch from pending Signatures to pending Sends

**Comment by VSaric** [comment]
> Changes in [confirm-page-container-navigation.component.js](https://github.com/MetaMask/metamask-extension/pull/17093/files#diff-b003f2a56b8e58f8ee673318a99652e8a205c3f1bd57b8b9384606d77b595dc8) can cause unapproved signature request to be opened from transaction page and visa versa.
> 
> An option could be creating separate component for signatures. This new component and `confirm-page-container-navigation.component.js` can be refactored to use same dumb component for UI rendering.

You are right @jpuri! I created separate component for signatures and this new component for signatures and `confirm-page-container-navigation.component.js` are refactored now. This two components are using same component for UI rendering (in this case `navigation.js`). 
I will appreciate your re-review on this and if there is some new suggestions please comment and I will address them. You can check latest changes [here](https://github.com/MetaMask/metamask-extension/pull/17093/commits/d929ddeb141fa82a86b9ef454f4d592f73a59056).

**Comment by VSaric** [comment]
> Some behaviour to double-check related to mixing Signatures with Sends/Deploys tx's following up @jpuri comment: after prompting for a couple of signatures without accepting them, if the Dapp now sends Send/Deploy tx's requests, MM starts to behave strangely. Issues:
> 
> * after some pending Signatures, you try a couple of Sends, and the top navigation displays now the number of pending Sends, but with the Signatures screens. Imagine you had 3 pending Signatures and 2 pending Sends: you'll see navigation for the Sends (2), but the screen for Signatures
> * if you then navigate to the Next page, you are now seeing the pending Sends with their navigation and the Signatures won't appears anymore until you reject/accept the pending Sends.
> 
>  sign-reqs-tx-reqs.mp4 
> I am not sure what should be the expect behaviour though:
> 
> * are we aiming to consider all Dapp API calls into one navigation flow? I think that would be good for the usability, but not sure how complex might be to implement, depending on how the code is at the moment
> * if we want to separate the different Dapp requests, we should fix the navigation mix and the clicking Next, making switch from pending Signatures to pending Sends

Hey @seaona, thank you for QA this. Yes, @jpuri was right and you are right. I created separate component for signatures and this new component for signatures and `confirm-page-container-navigation.component.js` are refactored now. This two components are using same component for UI rendering (in this case `navigation.js`).
With this latest changes there are no issues with mixing Signatures with Sends/Deploys tx's. Below is screencast with mixing Signatures with Sends/Deploys tx's with latest changes that I made in this PR:

https://user-images.githubusercontent.com/92527393/213481443-ce09ef92-9cb4-4184-926a-dffa60e550e1.mov

I will appreciate your re-review on this. Thanks!

**Comment by bschorchit** [comment]
@VSaric thank you for the update on the PR! Would it be possible to consolidate transactions and signatures in the same navigation flow following @seaona 1st suggestion instead of the 2nd? From the user perspective, it might be confusing to not be able to check the transactions requests until they reject or approve the signature ones. 

cc: @holantonela for visibility as it's design related

**Comment by jpuri** [comment]
Hey @VSaric : me and @bschorchit had chat about this and we decided that this approach will be better:
> are we aiming to consider all Dapp API calls into one navigation flow? I think that would be good for the usability, but not sure how complex might be to implement, depending on how the code is at the moment

It seems that this will increase your effort in the task and you almost need to re-do it. Can you plz check this.

We are aiming to have a single component for navigation which will show both messages and transactions and any other DAPP request.

**Comment by dzfjz** [comment]
Verified by QA

**Comment by VSaric** [comment]
> Hey @VSaric : code changes look great. Will be n ice to have some more test coverage also.

Hey @jpuri, I added more unit tests.

File: ui/components/app/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js [hunk 1]
```diff
@@ -42,7 +53,11 @@ const ConfirmPageContainerNavigation = () => {
   const onNextTx = (txId) => {
     if (txId) {
       dispatch(clearConfirmTransaction());
-      history.push(`${CONFIRM_TRANSACTION_ROUTE}/${txId}`);
+      history.push(
+        uncofirmedTransactions[txId]?.msgParams
+          ? `${CONFIRM_TRANSACTION_ROUTE}/${txId}${SIGNATURE_REQUEST_PATH}`
+          : `${CONFIRM_TRANSACTION_ROUTE}/${txId}`,
```
Comment (jpuri) [inline ui/components/app/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js:59]: I think if you route all request to `${CONFIRM_TRANSACTION_ROUTE}/${txId}` it takes care to re-route to specific page depending on request type and should work for any confirmation request in fact.

**Review by jpuri — APPROVED** [review]
Great work 👍

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**